pkg/server/slicerouter_config.go

```diff
@@ -281,58 +287,89 @@ func sliceRouterReconcileRoutingTable() error {
 	}
 
-	lastRoutingTableReconcileTime = time.Now()
+	_, routePresent := remoteSubnetRouteMap[remoteSubnet]
+	nextHopIpSlice := []*netlink.NexthopInfo{}
 
-	logger.GlobalLogger.Infof("RT reconciled at: %v", lastRoutingTableReconcileTime)
+	for i := 0; i < len(nextHopIPList); i++ {
 
-	_, routePresent := remoteSubnetRouteMap[remoteSubnet]
-	if routePresent && remoteSubnetRouteMap[remoteSubnet] == nextHopIP {
-		logger.GlobalLogger.Infof("Ignoring route add request. Route already installed. RemoteSubnet: %v, NextHop: %v",
-			remoteSubnet, nextHopIP)
-		return nil
-	}
+		gwObj := &netlink.NexthopInfo{Gw: net.ParseIP(nextHopIPList[i])}
+		nextHopIpSlice = append(nextHopIpSlice, gwObj)
 
-	if getSliceRouterDataplaneMode() == SliceRouterDataplaneVpp {
-		// If a route was previously installed for the remote subnet then we should
-		// delete it before adding a route with a new nexthop IP.
-		// VPP treats a route modify as a route add operation, creating multiple
-		// entries for a destination prefix and treating them as equal cost multipath
-		// routes.
-		// In our case, we should have only one route with the nexthop as the nsm IP on
-		// the slice gw pod connecting the remote subnet.
-		if remoteSubnetRouteMap[remoteSubnet] != "" {
-			err := vl3DeleteRouteInVpp(remoteSubnet, remoteSubnetRouteMap[remoteSubnet])
+		if routePresent && checkRouteAdd(remoteSubnetRouteMap[remoteSubnet], nextHopIPList[i]) {
+			logger.GlobalLogger.Infof("Ignoring route add request. Route already installed. RemoteSubnet: %v, NextHop: %v",
+				remoteSubnet, nextHopIPList[i])
+			return nil
+		}
+
+		if getSliceRouterDataplaneMode() == SliceRouterDataplaneVpp {
+			// If a route was previously installed for the remote subnet then we should
+			// delete it before adding a route with a new nexthop IP.
+			// VPP treats a route modify as a route add operation, creating multiple
+			// entries for a destination prefix and treating them as equal cost multipath
+			// routes.
+			// In our case, we should have only one route with the nexthop as the nsm IP on
+			// the slice gw pod connecting the remote subnet.
+			if len(remoteSubnetRouteMap[remoteSubnet]) != 0 {
+				err := vl3DeleteRouteInVpp(remoteSubnet, remoteSubnetRouteMap[remoteSubnet][i])
+				if err != nil {
+					logger.GlobalLogger.Errorf("Failed to delete route with old gw IP. RemoteSubent: %v, NextHop: %v",
+						remoteSubnet, remoteSubnetRouteMap[remoteSubnet][i])
+					return err
+				}
+			}
+			err := vl3InjectRouteInVpp(remoteSubnet, nextHopIPList[i])
 			if err != nil {
-				logger.GlobalLogger.Errorf("Failed to delete route with old gw IP. RemoteSubent: %v, NextHop: %v",
-					remoteSubnet, remoteSubnetRouteMap[remoteSubnet])
 				return err
 			}
 		}
-		err := vl3InjectRouteInVpp(remoteSubnet, nextHopIP)
-		if err != nil {
-			return err
+
+		if getSliceRouterDataplaneMode() == SliceRouterDataplaneKernel && i == len(nextHopIPList)-1 {
```

Comment:
This conditional looks convoluted. Inside a loop you should try to avoid checks on the index unless there is absolutely no other way. Why cant the call to inject route in kernel func be made at the place where you are checking if the route is present or not on line 311?